packages/edit-site/src/components/page-patterns/duplicate-menu-item.js

```diff
@@ -119,17 +143,18 @@ export default function DuplicateMenuItem( {
 				__( '%s (Copy)' ),
 				item.title
 			);
-
+			const patternCategory = await findOrCreateTerm( categoryId );
```

Comment:
This might cause two issues:
1. New patterns may be assigned to the generic "All patterns"
2. Only assigns the pattern to the current category rather than all the categories of the original pattern.

Reply:
I have pushed an update that uses the categories of the pattern rather than the current category listing